tests/storage/test_room_search.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2021 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import synapse.rest.admin
+from synapse.rest.client import login, room
+from synapse.storage.engines import PostgresEngine
+
+from tests.unittest import HomeserverTestCase
+
+
+class NullByteInsertionTest(HomeserverTestCase):
+    servlets = [
+        synapse.rest.admin.register_servlets_for_client_rest_resource,
+        login.register_servlets,
+        room.register_servlets,
+    ]
+
+    def test_null_byte(self):
+        """
+        Postgres/SQLite don't like null bytes going into the search tables. Internally
+        we replace those with a space.
+
+        Ensure this doesn't break anything.
+        """
+
+        # register a user and create a room, create some messages
+        self.register_user("alice", "password")
+        access_token = self.login("alice", "password")
+        room_id = self.helper.create_room_as("alice", True, "1", access_token)
+        body1 = "hi\u0000bob"
+        body2 = "another message"
+        body3 = "hi alice"
+
+        # send messages and ensure they don't cause an internal server
+        # error
+        resp1 = self.helper.send(room_id, body1, "1", access_token)
+        resp2 = self.helper.send(room_id, body2, "2", access_token)
+        resp3 = self.helper.send(room_id, body3, "3", access_token)
+        self.assertTrue("event_id" in resp1)
+        self.assertTrue("event_id" in resp2)
+        self.assertTrue("event_id" in resp3)
```

Comment:
We can simplify this a bit:

1. Remove some duplicated code.
2. Use longer variable names.
3. User `assertIn`.

```suggestion
        # Send messages and ensure they don't cause an error.
        for body in ["hi\u0000bob", "another message", "hi alice"]:
            response = self.helper.send(room_id, body, tok=access_token)
            self.assertIn("event_id", response)
```